src/tensor_api.cpp

```diff
@@ -48,6 +48,23 @@ extern "C" miopenStatus_t miopenSet4dTensorDescriptor(
     });
 }
 
+extern "C" miopenStatus_t miopenSet4dTensorDescriptorWithLayout(miopenTensorDescriptor_t tensorDesc,
```

Comment:
Why not make it `...Nd...` with `int* lens` and `int num_lens` parameters?

Reply:
The tensor api designed here follow the fashion of other miopen tensor api style.

Reply:
I second @DrizztDoUrden's opinion

Reply:
I don't think we should spawn 3d/4d/5d function signatures and implementation, when there is additional complexity in making just one. Later each of them would require either support, or to be implemented via an internal Nd function.

Reply:
Hi @DrizztDoUrden , these function has been changed to Nd already in some commits ago. Sorry for late inform.